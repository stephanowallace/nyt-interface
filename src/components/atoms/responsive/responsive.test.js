import React from 'react';
import { shallow } from 'enzyme';
import Constants from 'Constants';
import Responsive from '.';

describe('Responsive', () => {
  const child = ({ mode }) => <div>{mode}</div>;

  it('renders default state', () => {
    const component = shallow(<Responsive>{child}</Responsive>);
    expect(component).toMatchSnapshot();
  });

  it('clears timeout on resize', () => {
    const component = shallow(<Responsive>{child}</Responsive>);
    component.instance().timeout = 100;
    const clearTimeoutFunc = jest.spyOn(global, 'clearTimeout');
    const setTimeoutFunc = jest.spyOn(global, 'clearTimeout');
    global.innerWidth = 763;
    global.dispatchEvent(new Event('resize'));
    expect(clearTimeoutFunc).toHaveBeenCalledWith(100);
    expect(setTimeoutFunc).toHaveBeenCalled();
  });

  it('removes listener on unmount', () => {
    const component = shallow(<Responsive>{child}</Responsive>);
    const removeEventListenerFunc = jest.spyOn(global, 'removeEventListener');
    component.unmount();
    expect(removeEventListenerFunc).toHaveBeenCalled();
  });

  it('returns correct screen sizes', () => {
    const component = shallow(<Responsive>{child}</Responsive>);
    global.innerWidth = Constants.SCREEN_SIZES.EXTRA_SMALL - 1;
    expect(component.instance().getScreenMode()).toBe(Constants.SCREEN_SIZES.EXTRA_SMALL);
    global.innerWidth = Constants.SCREEN_SIZES.EXTRA_SMALL + 1;
    expect(component.instance().getScreenMode()).toBe(Constants.SCREEN_SIZES.SMALL);
    global.innerWidth = Constants.SCREEN_SIZES.SMALL + 1;
    expect(component.instance().getScreenMode()).toBe(Constants.SCREEN_SIZES.MEDIUM);
    global.innerWidth = Constants.SCREEN_SIZES.MEDIUM + 1;
    expect(component.instance().getScreenMode()).toBe(Constants.SCREEN_SIZES.LARGE);
  });
});
