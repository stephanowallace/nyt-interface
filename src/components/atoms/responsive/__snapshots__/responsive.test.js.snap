// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Responsive renders default state 1`] = `
<div>
  1200
</div>
`;
